### ISSUE TITLE
[False Negative] enable_error_codes is completely ignored if any specific code is unrecognised

### ISSUE THREAD

**Issue by ShadowLNC** [python/mypy#17648]
**Bug Report**

If I try to enable an error code which is unrecognised, then the entire `enable_error_codes` option is ignored.

**To Reproduce**

repro.py

```python
x = 3
x += "a"  # type: ignore
```

pyproject.toml

```toml
[project]
name = "my-project"
version = "0"
requires-python = ">=3.9"

[tool.mypy]
python_version = "3.9"
files = "."
enable_error_code = ["ignore-without-code", "not-exists-code"]

incremental = false
cache_dir = "/dev/null"
warn_unused_configs = true
```

**Expected Behavior**

Either mypy should refuse to run and exit nonzero due to unrecognised configuration, or it should act as though I never specified `not-exists-code`, like so:

```shell
repro.py:2: error: "type: ignore" comment without error code (consider "type: ignore[operator]" instead)  [ignore-without-code]
Found 1 error in 1 file (checked 1 source file)
```

**Actual Behavior**

```shell
pyproject.toml: [mypy]: enable_error_code: Invalid error code(s): not-exists-code
Success: no issues found in 1 source file
```

Exit code zero (success)

**Your Environment**

- Mypy version used: `mypy==1.11.1`, `mypy-extensions==1.0.0`, `typing_extensions==4.12.2` (empty Virtual Environment with `pip install mypy` as of today, `mypy --version` gives `mypy 1.11.1 (compiled: yes)`)
- Mypy command-line flags: none (`mypy repro.py`)
- Mypy configuration options from `mypy.ini` (and other config files): as above
- Python version used: Python 3.11.1 (Virtual Environment on macOS, Python installed from https://python.org )

**Comment by wyattscarpenter** [comment]
Seems to work on master now. Specifically via "refuse to run and exit nonzero"

```
uv run mypy --enable-error-code=ignore-without-code --enable-error-code=not-exists-code -c "#type: ignore"
mypy: error: Invalid error code(s): not-exists-code
Found 1 error in 1 file (errors prevented further checking)
```
(exit code 2)

```
uv run mypy --config=tmp.ini -c "#type: ignore"
mypy: error: Invalid error code(s): not-exists-code
Found 1 error in 1 file (errors prevented further checking)
```
(exit code 2)

```
uv run mypy --config=tmp.toml -c "#type: ignore"
mypy: error: Invalid error code(s): not-exists-code
Found 1 error in 1 file (errors prevented further checking)
```
(exit code 2)

----

Idk if we have any test coverage of this. I started writing the following in `test-data\unit\check-errorcodes.test`, and then realized I could just do the above and so I abandoned the below.

<details> <summary> some tests and what they printed when run </summary>

```
[case testOnlyInvalidErrorCodesIgnored]
-- Regression test for https://github.com/python/mypy/issues/17648
# flags: --enable-error-code=ignore-without-code --enable-error-code=not-exists-code
# type: ignore # E: "type: ignore" comment without error code (consider "type: ignore[operator]" instead)  [ignore-without-code]
```
↓
```
---------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------
Expected:
  main:2: error: "type: ignore" comment without error code (consider "type: ignore[operator]" instead)  [ignore-without-code] (diff)
Actual:
  mypy: error: Invalid error code(s): not-exists-code (diff)

Alignment of first line difference:
  E: main:2: error: "type: ignore" comment without error code (consider "type...
  A: mypy: error: Invalid error code(s): not-exists-code...
      ^
Update the test output using --update-data (implies -n0; you can additionally use the -k selector to update only specific tests)
======================================================================================================== short test summary info =========================================================================================================
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-errorcodes.test::testOnlyInvalidErrorCodesIgnored
================================================================================================== 1 failed, 13274 deselected in 1.89s ===================================================================================================

FAILED: testOnlyInvalidErrorCodesIgnored
```

```
[case testOnlyInvalidErrorCodesIgnored]
-- Regression test for https://github.com/python/mypy/issues/17648
# flags: --config=tmp/n.ini
# type: ignore # E: "type: ignore" comment without error code (consider "type: ignore[operator]" instead)  [ignore-without-code]
[file n.ini]
\[mypy]
enable_error_code = ignore-without-code,not-exists-code
```
↓
```
---------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------
Expected:
  main:2: error: "type: ignore" comment without error code (consider "type: ignore[operator]" instead)  [ignore-without-code] (diff)
Actual:
  mypy: error: Invalid error code(s): not-exists-code (diff)

Alignment of first line difference:
  E: main:2: error: "type: ignore" comment without error code (consider "type...
  A: mypy: error: Invalid error code(s): not-exists-code...
      ^
Update the test output using --update-data (implies -n0; you can additionally use the -k selector to update only specific tests)
======================================================================================================== short test summary info =========================================================================================================
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-errorcodes.test::testOnlyInvalidErrorCodesIgnored
================================================================================================== 1 failed, 13274 deselected in 1.92s ===================================================================================================

FAILED: testOnlyInvalidErrorCodesIgnored
```

</details>

**Comment by hauntsaninja** [comment]
Great! I'll go ahead and close this then
